### PR TITLE
feat: implement ERC20 token logic

### PR DESCRIPTION
**Pull request by estheroche**

### CODE REVIEW

File: src/bwc_erc20_token.cairo [hunk 1]
```diff
@@ -1,16 +1,215 @@
...
+    );
+}
+
 #[starknet::contract]
+mod ERC20 {
+    // importing necessary libraries
+    use starknet::ContractAddress;
+    use starknet::get_caller_address;
+    use starknet::contract_address_const; //similar to address(0) in Solidity
+    use zeroable::Zeroable;
```
Comment (sprtd) [inline src/bwc_erc20_token.cairo]: this contract is not compiling due to `use zeroable::Zeroable;`

import appropriately using: 
`use core::zeroable::Zeroable;`

File: src/bwc_erc20_token.cairo [hunk 2]
```diff
@@ -1,16 +1,215 @@
...
+    );
+    fn approve(ref self: TContractState, spender: ContractAddress, amount: u256);
+    fn increase_allowance(ref self: TContractState, spender: ContractAddress, added_value: u256);
+    fn decrease_allowance(
+        ref self: TContractState, spender: ContractAddress, subtracted_value: u256
+    );
+}
+
 #[starknet::contract]
+mod ERC20 {
```
Comment (sprtd) [inline src/bwc_erc20_token.cairo]: use `BWCERC20Token` as contract name

File: src/bwc_erc20_token.cairo
Comment (sprtd) [inline src/bwc_erc20_token.cairo:1]: @estheroche effect all the changes I suggested earlier
overall, you're doing great
keep going!

File: src/bwc_erc20_token.cairo [hunk 3]
```diff
@@ -1,16 +1,215 @@
...
+        }
     }
 
+    #[generate_trait]
+    impl HelperImpl of HelperTrait {
+        fn transfer_helper(
+            ref self: ContractState,
+            sender: ContractAddress,
+            recipient: ContractAddress,
+            amount: u256
```
Comment (sprtd) [inline src/bwc_erc20_token.cairo:174]: add validation logic to check that `amount` can only be `<=` sender's token balance
meaning you must add the ffg logic
```
let sender_balance = self.balances.read(sender)
assert(amount <= sender_balance, 'insufficient balance');
```

**Review by sprtd — APPROVED** [review]

File: src/bwc_erc20_token.cairo
Comment (estheroche) [reply src/bwc_erc20_token.cairo:1]: thank you sir and thank for teaching to become a better developer.

**Pull request merged**